File: scratch/gopurs-adt-reuse-validation-20260916/before-pure/src/Test/FibFFICheatcode.go
package Test_FibFFICheatcode


func fib_cheatcode(n int) int {
	if n == 0 {
		return 0
	}
	if n == 1 {
		return 1
	}
	return fib_cheatcode(n-1) + fib_cheatcode(n-2)
}

func RunFibFFICheatcode(limit int) int {
	dummy := limit
	return (fib_cheatcode(dummy))
}

File: scratch/gopurs-adt-reuse-validation-20260916/before-pure/src/Test/FibFFICheatcode.cc
#include "purescript.h"
FOREIGN_BEGIN( Test_FibFFICheatcode )
exports["runFibFFICheatcode"] = [](const boxed& in) -> boxed {
    const int n = unbox<int>(in);
    if (n == 0) return 0;
    int previous = 0;
    int current = 1;
    for (int i = 1; i < n; ++i) {
        const int next = previous + current;
        previous = current;
        current = next;
    }
    return current;
};
FOREIGN_END
